webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/tracings/volume/VolumeSegmentIndexBuffer.scala

```diff
@@ -47,4 +54,20 @@ class VolumeSegmentIndexBuffer(tracingId: String, volumeSegmentIndexClient: Foss
       }
     } yield ()
 
+  private def getFallback(segmentId: Long, mag: Vec3Int, mappingName: Option[String])(implicit ec: ExecutionContext) = {
+    val key = segmentIndexKey(tracingId, segmentId, mag)
+    for {
+      fossilDbData <- volumeSegmentIndexClient
+        .get(key, Some(version), mayBeEmpty = Some(true))(fromProtoBytes[ListOfVec3IntProto])
+        .map(_.value)
+        .fillEmpty(ListOfVec3IntProto.of(Seq()))
+      layerData <- fallbackLayer match {
+        case Some(layer) if fossilDbData.length == 0 =>
+          remoteDatastoreClient.querySegmentIndex(layer, segmentId, mag, mappingName, userToken)
+        case _ => Fox.successful(Seq.empty)
+      }
+      combined = fossilDbData.values.map(vec3IntFromProto) ++ layerData
```

Comment:
Am I reading this correctly that combined will in the end have either fossilDbData values or layerData values but not both? In this case I’m not super sure about the `++` operator. It should at least be explained in a comment. Maybe a getOrElse cascade can be used instead, to make the intent clearer

Reply:
Made it a little clearer now